src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -633,12 +644,155 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
+            .conceptRelationships(
+                List.of(
+                    "fpk_drug_concept",
+                    "fpk_drug_type_concept",
+                    "fpk_drug_route_concept",
+                    "fpk_drug_concept_s")));
+    tableMap.put(
+        "Measurement",
+        new SnapshotBuilderTable()
+            .datasetTableName("measurement")
+            .personRelationship("fpk_person_measurement")
+            .conceptRelationships(
+                List.of(
+                    "fpk_measurement_concept",
+                    "fpk_measurement_unit",
+                    "fpk_measurement_concept_s",
+                    "fpk_measurement_value",
+                    "fpk_measurement_type_concept",
+                    "fpk_measurement_operator")));
+    tableMap.put(
+        "Visit",
+        new SnapshotBuilderTable()
+            .datasetTableName("visit_occurrence")
+            .personRelationship("fpk_person_visit")
+            .conceptRelationships(
+                List.of(
+                    "fpk_visit_preceding",
+                    "fpk_visit_concept_s",
+                    "fpk_visit_type_concept",
+                    "fpk_visit_concept",
+                    "fpk_visit_discharge")));
+    tableMap.put(
+        "Device",
+        new SnapshotBuilderTable()
+            .datasetTableName("device_exposure")
+            .personRelationship("fpk_person_device")
+            .conceptRelationships(
+                List.of("fpk_device_concept", "fpk_device_concept_s", "fpk_device_type_concept")));
+    tableMap.put(
+        "Condition",
+        new SnapshotBuilderTable()
+            .datasetTableName("condition_occurrence")
+            .personRelationship("fpk_person_condition")
+            .conceptRelationships(
+                List.of(
+                    "fpk_condition_concept",
+                    "fpk_condition_type_concept",
+                    "fpk_condition_status_concept",
+                    "fpk_condition_concept_s")));
+    tableMap.put(
+        "Procedure",
+        new SnapshotBuilderTable()
+            .datasetTableName("procedure_occurrence")
+            .personRelationship("fpk_person_procedure")
+            .conceptRelationships(
+                List.of(
+                    "fpk_procedure_concept",
+                    "fpk_procedure_concept_s",
+                    "fpk_procedure_type_concept",
+                    "fpk_procedure_modifier")));
+    tableMap.put(
+        "Observation",
+        new SnapshotBuilderTable()
+            .datasetTableName("observation")
+            .personRelationship("fpk_person_observation")
+            .conceptRelationships(
+                List.of(
+                    "fpk_observation_concept",
+                    "fpk_observation_concept_s",
+                    "fpk_observation_unit",
+                    "fpk_observation_qualifier",
+                    "fpk_observation_type_concept",
+                    "fpk_observation_value")));
```

Comment:
so once this information is in the settings, the custodian will need to define this information twice is that right? Once on the dataset for all tables, and then again on the snapshot settings for each table. If that's the case it makes me want to revisit the auto generating settings once this upcoming settings work is done.

Reply:
Yes, that's correct. I think autogenerating the relationships would make a lot of sense for future improvement.

Reply:
Post-MVP ticket: https://broadworkbench.atlassian.net/browse/DC-1168